frontend/src/lib/MlmdUtils.test.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  Api,
+  Context,
+  GetContextByTypeAndNameRequest,
+  GetContextByTypeAndNameResponse,
+} from '@kubeflow/frontend';
+import { expectWarnings, testBestPractices } from 'src/TestUtils';
+import { Workflow, WorkflowSpec, WorkflowStatus } from 'third_party/argo-ui/argo_template';
+import { getRunContext } from './MlmdUtils';
+
+testBestPractices();
+
+const WORKFLOW_NAME = 'run-st448';
+const WORKFLOW_EMPTY: Workflow = {
+  metadata: {
+    name: WORKFLOW_NAME,
+  },
+  // there are many unrelated fields here, omit them
+  spec: {} as WorkflowSpec,
+  status: {} as WorkflowStatus,
+};
+
+const V2_CONTEXT = new Context();
+V2_CONTEXT.setName(WORKFLOW_NAME);
+V2_CONTEXT.setType('kfp.PipelineRun');
+
+const TFX_CONTEXT = new Context();
+TFX_CONTEXT.setName('run.run-st448');
+TFX_CONTEXT.setType('run');
+
+const V1_CONTEXT = new Context();
+V1_CONTEXT.setName(WORKFLOW_NAME);
+V1_CONTEXT.setType('KfpRun');
+
+describe('MlmdUtils', () => {
+  describe('getRunContext', () => {
+    it('gets KFP v2 context', async () => {
+      mockGetContextByTypeAndName([V2_CONTEXT]);
+      const context = await getRunContext({
+        ...WORKFLOW_EMPTY,
+        metadata: {
+          ...WORKFLOW_EMPTY.metadata,
+          annotations: { 'pipelines.kubeflow.org/v2_pipeline': 'true' },
+        },
+      });
+      expect(context).toEqual(V2_CONTEXT);
+    });
+
+    it('gets TFX context', async () => {
+      mockGetContextByTypeAndName([TFX_CONTEXT, V1_CONTEXT]);
```

Comment:
Question: Should we also add `V2_CONTEXT` to this mock? We should expect this test to return TFX_CONTEXT because WORKFLOW_EMPTY doesn't have v2 annotation.

Reply:
Therefore I don't feel whether adding `V2_CONTEXT` makes any difference.

Reply:
I agree that it won't affect the test result. V2_CONTEXT condition `type: 'kfp.PipelineRun'` is examined before reaching TFX_CONTEXT in the production code, so I am only suggesting for the completeness of mocking.